### DISCUSSION TITLE
Message API 갯수 카운트 아이디어

### DISCUSSION THREAD

**Discussion by nyeoni** [GhostPangPang/GhostPong#118] [Ideas]
### 발단
- 기존에 `message_id` 가 `auto_increment` 였는데, 이 부분을 개선하여 안읽은 메세지 갯수 카운트 하는 아이디어가 생각남

### 아이디어
- `message_id` 를 `string` 으로 `${friend-id}-${message_index}` 형식으로 해결할 수 있음
- db `message_view` 테이블에 마지막으로 읽은 읽자 정보를 저장하니까, 이 date 로 `message` 테이블의 데이터를 조회하고 current_date 의 데이터를 조회해서 `message_id` 부분의 `message_index` 를 빼는 방식으로 카운트를 할 수 있음

### 주의사항
- 위의 로직은 `/friend` 엔드포인트에서 이루어져야 하며 추가적으로 2개정도의 db 쿼리가 생김
- `message` 테이블의 `id` 를 바꿔야하는 작업이 들어감

@jis-kim @anso33 @Kimhan-nah @aLVINlEE9  의견주세용

**Comment by jis-kim** [comment]
굿아이디어입니다! 스프린트 2 종료 후 변경 고려해보겠습니다. 실시간으로 메세지가 온 이벤트를 웹소켓으로 쏴 줄 때도 생성된 메세지 id 를 전송해주면 계산이 수월할 것 같아요. 변경하게 되면 `message_view` 에 마지막으로 읽은 시간 대신 마지막으로 읽은 메세지 아이디를 저장하는 것도 괜찮을 것 같습니다.

**Reply by jis-kim** [reply]
만약 이 방안이 적용되면 
- #104 도 어느정도 해결됩니다.
